osu.Game/Overlays/Chat/ChannelList/ChannelList.cs

```diff
@@ -0,0 +1,139 @@
+                    Child = new FillFlowContainer
+                    {
+                        Direction = FillDirection.Vertical,
+                        RelativeSizeAxes = Axes.X,
+                        AutoSizeAxes = Axes.Y,
+                        Children = new Drawable[]
+                        {
+                            publicChannelFlow = new ChannelListItemFlow("CHANNELS"),
+                            new ChannelListSelector
+                            {
+                                Margin = new MarginPadding { Bottom = 10 },
+                            },
+                            privateChannelFlow = new ChannelListItemFlow("DIRECT MESSAGES"),
+                        },
+                    },
+                },
+            };
+        }
+
+        public void AddChannel(Channel channel)
+        {
+            if (channelMap.ContainsKey(channel))
+                return;
+
+            ChannelListItem item = new ChannelListItem(channel);
+            item.OnRequestSelect += chan => OnRequestSelect?.Invoke(chan);
+            item.OnRequestLeave += chan => OnRequestLeave?.Invoke(chan);
+
+            ChannelListItemFlow flow = getFlowForChannel(channel);
+            channelMap.Add(channel, item);
+            flow.Add(item);
+        }
+
+        public void RemoveChannel(Channel channel)
+        {
+            if (!channelMap.ContainsKey(channel))
+                return;
+
+            ChannelListItem item = channelMap[channel];
+            ChannelListItemFlow flow = getFlowForChannel(channel);
+
+            channelMap.Remove(channel);
+            flow.Remove(item);
+        }
+
+        public ChannelListItem GetItem(Channel channel)
+        {
+            if (!channelMap.ContainsKey(channel))
+                throw new ArgumentOutOfRangeException();
+
+            return channelMap[channel];
+        }
+
+        private ChannelListItemFlow getFlowForChannel(Channel channel)
+        {
+            switch (channel.Type)
+            {
+                case ChannelType.Public:
+                    return publicChannelFlow;
+
+                case ChannelType.PM:
+                    return privateChannelFlow;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private class ChannelListItemFlow : FillFlowContainer
+        {
+            public ChannelListItemFlow(string label)
+            {
+                Direction = FillDirection.Vertical;
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+
+                Add(new OsuSpriteText
+                {
+                    Text = label,
+                    Margin = new MarginPadding { Left = 18, Bottom = 5 },
+                    Font = OsuFont.Torus.With(size: 12, weight: FontWeight.SemiBold),
+                });
+            }
+        }
+    }
+
+    public enum ChannelSelectorState
+    {
+        Visibile,
+        Hidden,
+    }
```

Comment:
This feels a touch like overkill? Like this is just a boolean with a flipped default, essentially. I like strong typing but not sure this achieves much at all.

(Also `Visibile` has a typo in it.)

Reply:
This stemmed from the decision to resolve instead of passing via bindtarget to every child element since caching a `BindableBool` in the hierarchy seemed not good. If it's preferred to always use bindtarget for all the child items then I would just use the `BindableBool` instead.